src/table/settings/settings.ini

```diff
@@ -2506,7 +2506,7 @@ cat      = SC_BASIC
 var      = gui.scroll_mode
 type     = SLE_UINT8
 flags    = SF_NOT_IN_SAVE | SF_NO_NETWORK_SYNC | SF_GUI_DROPDOWN
-def      = 0
+def      = 2
```

Comment:
Could you make this change happen only for emscripten? As mentioned earlier, check for `APPLE` to see examples how. Or check back in if this wasn't enough breadcrumbs :D

Reply:
Sounds good. I had a look at the way other settings handle this, and it seems that the setting entry would have to be [duplicated in its entirety](https://github.com/OpenTTD/OpenTTD/blob/003b6a0c85157059d0339dbfa551db7ec3b40cd9/src/table/settings/misc_settings.ini#L68-L80)?

The only solution I see is to use normal preprocessor directives in the `pre-amble` section to `#define DEFAULT_SCROLL_MODE` based on whether EMSCRIPTEN is defined or not, and then set `def = DEFAULT_SCROLL_MODE`. That seems just as ugly.

Thoughts?

Reply:
I do not think you need a new define for this, at least, that is not what we do with MacOS. Something like this:

```
[SDTC_VAR]
ifndef = __EMSCRIPTEN__
var      = gui.scroll_mode
type     = SLE_UINT8
flags    = SF_NOT_IN_SAVE | SF_NO_NETWORK_SYNC | SF_GUI_DROPDOWN
def      = 0
min      = 0
max      = 3
str      = STR_CONFIG_SETTING_SCROLLMODE
strhelp  = STR_CONFIG_SETTING_SCROLLMODE_HELPTEXT
strval   = STR_CONFIG_SETTING_SCROLLMODE_DEFAULT
cat = SC_BASIC

[SDTC_VAR]
ifdef = __EMSCRIPTEN__
var      = gui.scroll_mode
type     = SLE_UINT8
flags    = SF_NOT_IN_SAVE | SF_NO_NETWORK_SYNC | SF_GUI_DROPDOWN
def      = 2
min      = 0
max      = 3
str      = STR_CONFIG_SETTING_SCROLLMODE
strhelp  = STR_CONFIG_SETTING_SCROLLMODE_HELPTEXT
strval   = STR_CONFIG_SETTING_SCROLLMODE_DEFAULT
cat = SC_BASIC
```

Might do the trick. It is far from pretty, but atm we do not really have a cleaner way :D

PS: code above is untested, so please use it with care :)

Reply:
Looks like this works; I tested on both Ubuntu 18.04 (by the way, had to manually install and force use of GCC 8 because apparently `<charconv>` isn't available on GCC 7) and Emscripten.